File: Script Statiche/titoloPlot.m
function [titolo, W , L , type] = titoloPlot(nomeCartella)
    %titolo: il titolo del plot (es NMOS 100/0.030)
    %W : la largezza del canale in um (es 100)
    %L : la lunghezza del canale in um (es 0.03)
    %type: il tipo del dispositivo

    type = nomeCartella(1);
    % Dividi la stringa usando il carattere "-"
    tokens = strsplit(nomeCartella, '-');
    
    % Estrai i componenti
    tipoTransistor = type + "MOS";
    larghezzaCanale = tokens{2};
    lunghezzaCanale = tokens{3};
    
    % Converti la larghezza del canale in formato corretto
    lunghezzaCanale = sprintf('%.3f', str2double(lunghezzaCanale) / 1000);
    
    % Costruisci la nuova stringa
    titolo = sprintf('%s %s/%s', tipoTransistor, larghezzaCanale, lunghezzaCanale);
    
    %sono in micro-metri
    W = str2double(larghezzaCanale);
    L = str2double(lunghezzaCanale);

end

File: Script Python Statiche/funzioni_ausiliarie/titolo_plot.py
from pathlib import Path

def titolo_plot(nome_cartella):
    """
    Restituisce:
    - titolo: stringa per il titolo del plot (es. 'NMOS 100/0.030')
    - W: larghezza del canale in µm (es. 100)
    - L: lunghezza del canale in µm (es. 0.03)
    - tipo: tipo del dispositivo ('N' o 'P')
    """

    nome = Path(nome_cartella).name
    tipo = nome[0]
    tokens = nome.split("-")

    try:
        tipo_transistor = f"{tipo}MOS"
        larghezza = float(tokens[1])
        lunghezza = float(tokens[2]) / 1000  # da nm a µm
        titolo = f"{tipo_transistor} {int(larghezza)}/{lunghezza:.3f}"
        return titolo, larghezza, lunghezza, tipo
    except Exception as e:
        raise ValueError("Errore nel parsing del nome della cartella, assicurati che sia nel formato 'N5-600-180'") from e
